sw/device/silicon_creator/lib/drivers/watchdog_functest.c

```diff
@@ -24,21 +23,32 @@
 
 // Tests that we can pet the watchdog and avoid a reset.
 rom_error_t watchdog_pet_test(void) {
-  watchdog_init(500);
+  // Set watchdog bite threshold to 5ms.
+  uint32_t threshold = 5 * (kClockFreqAonHz / 1000);
+  LOG_INFO("threshold = %d", threshold);
+  watchdog_configure(threshold, kHardenedBoolTrue);
+
   for (size_t i = 0; i < 10; ++i) {
-    LOG_INFO("watchdog = %x", watchdog_get());
     watchdog_pet();
-    usleep(5000);
+
+    // Sleep for 1ms.
+    usleep(1 * 1000);
   }
-  watchdog_init(0);
+  watchdog_disable();
   return kErrorOk;
 }
 
 // Tests that if we neglect the dog, it will bite and reset the chip.
 rom_error_t watchdog_bite_test(void) {
-  watchdog_init(1);
-  usleep(11000);
-  watchdog_init(0);
+  // Set watchdog bite threshold to 5ms.
+  uint32_t threshold = 5 * (kClockFreqAonHz / 1000);
+  LOG_INFO("threshold = %d", threshold);
+  watchdog_configure(threshold, kHardenedBoolTrue);
```

Comment:
Don't you think it's worth to have a test with `kHardenedBoolFalse` in the secont argument? Maybe a unittest.

Reply:
Yeah I think you're right, I've added an additional func test case.